### PR TITLE
move engine.io-client to dev dependency

### PR DESCRIPTION
**Pull request by camallen**
it's not used directly in this system see https://github.com/zooniverse/sugar/issues/82 for details

Staging branch URL: https://pr-6097.pfe-preview.zooniverse.org

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Tests fail with
```
Error: Missing required `engine.io-client` module. Please run `npm install --save engine.io-client`
```
which is thrown here in `primus.client()`.
https://github.com/zooniverse/Sugar-Client/blob/11093a1935c9bb70e7cf8021c4aa7673cc4d0ccd/primus.js#L2226-L2234

**Comment by camallen** [comment]
> Tests fail with
> 
> ```
> Error: Missing required `engine.io-client` module. Please run `npm install --save engine.io-client`
> ```
> 
> which is thrown here in `primus.client()`. https://github.com/zooniverse/Sugar-Client/blob/11093a1935c9bb70e7cf8021c4aa7673cc4d0ccd/primus.js#L2226-L2234

Yep - i don't see this when using `npm start` locally only in the test env 🤔

**Comment by eatyourgreens** [comment]
https://pr-6907.pfe-preview.zooniverse.org/ gives a 404 error too.

**Comment by camallen** [comment]
> https://pr-6907.pfe-preview.zooniverse.org/ gives a 404 error too.

My mistake, I've already fixed the URL to reflect the PR number #6097 - try https://pr-6097.pfe-preview.zooniverse.org/

**Comment by eatyourgreens** [comment]
I can't remember where the Sugar client is used in the code, but I can see the active user list on Talk. I think that comes from Sugar (user accounts actively subscribed to the project channel?)
https://pr-6097.pfe-preview.zooniverse.org/projects/douglassday/transcribe-colored-conventions/talk?env=production

So it seems to be working, I think.

**Comment by eatyourgreens** [comment]
Ignore that, 'Active Users' uses the Sugar JSON-API client, not Primus.
https://github.com/zooniverse/Panoptes-Front-End/blob/ce7ec636e21dd194c92b3ee9447ad549469639eb/app/talk/active-users.jsx#L4

**Comment by camallen** [comment]
the sugar system is being used on the talk index pages
- presence subscription - https://github.com/zooniverse/Panoptes-Front-End/blob/ce7ec636e21dd194c92b3ee9447ad549469639eb/app/talk/index.cjsx#L27
- presence unsubscription - https://github.com/zooniverse/Panoptes-Front-End/blob/ce7ec636e21dd194c92b3ee9447ad549469639eb/app/talk/index.cjsx#L34

**Comment by eatyourgreens** [comment]
Yeah, I'm just not sure what I'm supposed to be looking for in order to test that it's working. I'm not seeing any errors on staging though, so tentatively it seems to be working.

**Comment by eatyourgreens** [comment]
Anyhow, I guess the `engine.io-client` package was installed specifically to get the tests to pass. I'm not sure what's up with that.

**Comment by camallen** [comment]
> Yeah, I'm just not sure what I'm supposed to be looking for in order to test that it's working. I'm not seeing any errors on staging though, so tentatively it seems to be working.

I don't know why the test env is different from the dev / prod env which is wild. If we need the client lib installed for the CI system then we can move it to a dev dependency instead.

**Comment by eatyourgreens** [comment]
Tests are failing because `engine.io-client` can't be resolved here, which I assume is because it isn't found in the `node_modules` tree.
https://github.com/zooniverse/Sugar-Client/blob/11093a1935c9bb70e7cf8021c4aa7673cc4d0ccd/primus.js#L2226

I'm not really following how that gets resolved in the Webpack build, which does work.

**Comment by camallen** [comment]
I've moved the `engine.io-client` to a PFE dev dependency to make the CI test runs work.

File: package-lock.json [hunk 1]
```diff
@@ -2323,7 +2323,7 @@
     },
     "babel-plugin-syntax-jsx": {
       "version": "6.18.0",
-      "resolved": "https://registry.npmjs.org/babel-plugin-syntax-jsx/-/babel-plugin-syntax-jsx-6.18.0.tgz",
+      "resolved": "http://registry.npmjs.org/babel-plugin-syntax-jsx/-/babel-plugin-syntax-jsx-6.18.0.tgz",
```
Comment (eatyourgreens) [inline package-lock.json:2326]: This is a slightly suspicious change.

**Review by eatyourgreens — APPROVED** [review]
LGTM. Not sure why that one package in the lock file has change to load over HTTP instead of HTTPS.

**Pull request merged**